### ISSUE TITLE
Farm: Unlock New Berries not behaving properly

### ISSUE THREAD

**Issue by Sithishade** [Farigh/pokeclicker-automation#131]
### Describe the bug
The "Focus on unlocking plots and new berries" feature is not working for me.

### Steps to Reproduce
1. Have unlocked every berry up to Chople/Kebia/Shuca.
2. Turn off the "Optimize oak item loadout" feature under Focus, just to be safe.
3. Turn on "Focus on unlocking plots and new berries", as well as "Update oak item loadout when needed".
4. Turn on the Farming feature.

### Expected behavior
The tool will equip the Blaze Cassette and plant a full set of Spelon berries, in an attempt to mutate a Chople berry.

### Actual behavior
The tool plants a full set of Cheri berries.

### Additional context
Most recent non-beta version of Google Chrome (105.0.5195.102, 64 bit). Most recent version of tool (confirmed that I have the type priority setting in the hatchery, which is the most recent commit). Unsure if this issue is only with Oak Item loadout changes for the Farm--the tool _has_ unlocked Coba through Kasib, though has not skipped past the three aforementioned berries to Haban, which requires no loadout changes, as I'd expect if that were the only issue. I've manually unlocked Kebia and Shuca right now, and have a set of Spelon going manually to unlock Chople. I can provide a save file, if that is useful, or any other information you may need (though I think all the relevant settings have been set properly as above, unless "Update oak item loadout when needed" requires "Optimize oak item loadout" to be set under Focus, which would seem contradictory).

**Comment by Farigh** [comment]
Are you, by any chance, missing the Kasib berry ?

Berries are not unlocked in the same order as their ID. this is done to try unlocking easier berries first.

<details>
<summary>Here is the order they are unlocked in: (click to expend)</summary>

 ◈ Gen 1 unlocks:
    ✓ Unlock plot 7
    ✓ Unlock plot 13
    ✓ Unlock plot 17
    ✓ Unlock plot 11
    ✓ Unlock plot 6
    ✓ Unlock plot 8 
    ✓ Unlock plot 18 
    ✓ Unlock plot 16 
    ✓ Gen 1 berry gathering
  
  ◈ Gen 2 unlocks:
    ✓ Unlock Persim berry
    ✓ Unlock plot 2 
    ✓ Unlock Razz berry 
    ✓ Unlock plot 14 
    ✓ Unlock Bluk berry 
    ✓ Unlock plot 22 
    ✓ Unlock Nanab berry
    ✓ Unlock plot 10
    ✓ Unlock Wepear berry
    ✓ Unlock plot 3 (1 ms)
    ✓ Unlock Pinap berry
    ✓ Unlock plot 19
    ✓ Unlock Figy berry
    ✓ Unlock plot 21
    ✓ Unlock Wiki berry
    ✓ Unlock plot 5
    ✓ Unlock Mago berry 
    ✓ Unlock plot 1 
    ✓ Unlock Aguav berry 
    ✓ Unlock plot 9
    ✓ Unlock Iapapa berry
    ✓ Unlock plot 23
    ✓ Gen 2 berry gathering
  
  ◈ Gen 3 unlocks:
    ✓ Unlock Pomeg berry 
    ✓ Unlock plot 15 
    ✓ Unlock Kelpsy berry 
    ✓ Unlock plot 0 
    ✓ Unlock Qualot berry
    ✓ Unlock plot 4 
    ✓ Unlock Hondew berry 
    ✓ Unlock plot 24
    ✓ Unlock Grepa berry
    ✓ Unlock plot 20 
    ✓ Unlock Tamato berry 
    ✓ Unlock Cornn berry 
    ✓ Unlock Magost berry
    ✓ Unlock Rabuta berry
    ✓ Unlock Nomel berry
    ✓ Gen 3 next steps needed berry gathering
    ✓ Unlock Spelon berry
    ✓ Unlock Pamtre berry
    ✓ Unlock Watmel berry
    ✓ Unlock Durin berry 
    ✓ Unlock Belue berry
    ✓ Gen 3 berry gathering 
  
  ◈ Gen 4 unlocks:
    ✓ Unlock Occa berry 
    ✓ Unlock Coba berry 
    ✓ Unlock Passho berry
    ✓ Unlock Wacan berry
    ✓ Unlock Rindo berry 
    ✓ Unlock Yache berry 
    ✓ Unlock Payapa berry
    ✓ Unlock Tanga berry 
    ✓ Unlock Kasib berry 
    ✓ Unlock Haban berry
    ✓ Unlock Colbur berry
    ✓ Unlock Roseli berry 
    ✓ Unlock Shuca berry
    ✓ Unlock Charti berry 
    ✓ Unlock Babiri berry 
    ✓ Unlock Chople berry 
    ✓ Wait for plots to be emptied
    ✓ Unlock Chilan berry > step 1
    ✓ Unlock Chilan berry > step 2
    ✓ Unlock Kebia berry
  
  ◈ Gen 5 unlocks:
    ✓ Unlock Micle berry
    ✓ Unlock Custap berry
    ✓ Unlock Jaboca berry
    ✓ Unlock Rowap berry 
    ✓ Unlock Liechi berry 
    ✓ Unlock Ganlon berry 
    ✓ Unlock Kee berry
    ✓ Unlock Salac berry 
    ✓ Unlock Petaya berry
    ✓ Unlock Maranga berry
    ✓ Unlock Apicot berry
    ✓ Unlock Lansat berry 
    ✓ Unlock Starf berry 
    
  ◈ Bonus berries:
    ✓ Unlock Lum berry
    ✓ Unlock Enigma berry
  
</details>

As you can see, the Kasib berry (which requires a full Cheri setup) is unlocked way before Chople

**Comment by Farigh** [comment]
However, this requires the berry to wither, I'll fix this point.

**Comment by Sithishade** [comment]
Farigh,

I have unlocked the Kasib berry but do not have 4 of them, so that makes sense! I assumed the Farming module was reverting to its default "plant selected berry and harvest it" behavior, as when I opened the Farming module the first time I noticed the problem, Cheri was selected. It kept harvesting the Cheri berries too, so I figured it wouldn't have anything to do with new berry mutation, but clearly that's a separate issue. Glad this report was useful for something else at least? Closing this issue, and thanks for your quick response!
